test/plugins/Asset.test.ts

```diff
@@ -505,6 +515,94 @@ describe('Assets contracts #fast', () => {
       expect(cachedAtTime).to.eq(0)
     })
 
+    it('Should handle tokens being out on trade for RTokenAsset', async () => {
+      // Summary:
+      // - Run a dutch auction that does not fill
+      // - Run a batch auction that fills for partial volume
+      // - Run a dutch auction that fills for full volume
+
+      const low0 = fp('0.99')
+      const low1 = bn('975344098811881188') // after a 50% basket change
+      const low2 = bn('975343128415841584') // after batch auction at half volume
```

Comment:
low price falls? isn't it an invariant that this never happens?

Reply:
Every increase of `basketsHeld.bottom` (ie a successful round of purchasing full baskets) raises `RTokenAsset.price().low`, but there are small drops as the full baskets are purchased.

Reply:
I've changed a comment in commit aecbf64c16ba2191469d6eb152da1b6d33d12a26 to reflect this. Previous comment was actually technically wrong.

Reply:
ok i've looked deeper at the code and this makes sense to me now, thanks for the explanation